bin/main.ml

```diff
@@ -29,23 +27,16 @@ module Spec = struct
 
   let inode = Inter.Val.to_concrete
 
+  module I = Irmin.Hash.Typed (Store.Hash) (Inter.Val)
+
   let of_t (t : Inter.Val.t) =
-    let inode =
-      if Inter.Val.length t <= Conf.stable_hash then None else Some (inode t)
-    in
-    let bt = Inter.Val.to_bin t in
-    {
-      hash = Inter.Elt.hash bt;
-      bindings = List.map entry (Inter.Val.list t);
-      inode;
-    }
+    let stable = Inter.Val.stable t in
+    let inode = if stable then None else Some (inode t) in
+    let hash = I.hash t in
+    { hash; bindings = List.map entry (Inter.Val.list t); inode }
```

Comment:
Is this a simplification or did it have an impact on the bug? On my test files I didn't make this change, just changed 

```ocaml
module N = Irmin.Hash.Typed (Store.Hash) (Inter.Val)
```

deleted the Node module and then I generated the dataset and checked and it worked.

Reply:
yes it's a simplification - I'll revert it to make things clearer.